internal/resources/pods.go

```diff
@@ -427,10 +427,13 @@ func buildSmbdCtrs(
 	ctrs := []corev1.Container{}
 	ctrs = append(ctrs, buildSmbdCtr(planner, env, vols))
 	metaOnlyVols := vols.exclude(tagData)
-	// insert a container to watch for changes to the config json
-	// and apply those changes to samba
+	// Insert a container to watch for changes to the config json
+	// and apply those changes to samba.
+	// We need to pass data vols to the config update container as it is
+	// responsible for creating missing dirs and setting perms when new shares
+	// appear.
```

Comment:
I have to admit that I am not yet 100% confident with the current flow of actions from various containers. But still does that mean it is a bug? Or is it like a fix for future when configuration gets updated for additional shares?

Reply:
Mostly the latter. I have a very large WIP series of patches in #240 where I originally found the problem. The problem does exist in the current version in the case where sambacc is supposed to change permissions, but it's such a rare case that it is hard to hit. With the changes in the new PR it's very obvious because it fails to add new share subdirs to the PVC.

Reply:
Thanks for the clarification.